SUSE/scf
# Generate chart for SCF with cf-operator

## Description

Added a custom Bazel rule that downloads a pinned version of Helm to be used in a second custom rule that packages the SCF Helm chart.

Also removed cf-deployment from the source tree and added it as a Bazel dependency. The cf-deployment.yml file is appended to the final generated Helm chart.

The `Chart.yaml` appVersion and version got the values 0, as the packaging step will add the correct values. Those should be set when the `helm_package` rule is used. It is being used at `//deploy/helm/scf:chart`.

## Test plan

- Build the final chart package archive:
```
bazel build //deploy/helm/scf:chart
```
- Verify that it produced `bazel-bin/deploy/helm/scf/scf-3.0.0.tgz`.
- Verify that it includes the files under `deploy/helm/scf/` and the `cf-deployment.yml` under `scf/assets/cf-deployment.yml` in the `.tgz` file:
```
tar tvf bazel-bin/deploy/helm/scf/scf-3.0.0.tgz
```
